### ISSUE TITLE
Deep populate (Nested Eager Loading) only poplating the first element of the slice

### ISSUE THREAD

**Issue by Gacnt** [jinzhu/gorm#796]
```
package main

import (
	"log"

	_ "github.com/go-sql-driver/mysql"
	"github.com/jinzhu/gorm"
)

var db gorm.DB

type User struct {
	gorm.Model
	Username     string `sql:"not null;unique"`
	Email        string `sql:"not null;unique"`
	PasswordHash string `sql:"not null;unique"`
	UserLevel    int    `sql:"not null;default:1"`
	Characters   []Characters
	MainPosts    []ForumMainThreadsPosts
	ReplyPosts   []ForumMainPostsReplies
}

// Characters is a table in the database
type Characters struct {
	gorm.Model
	UserID             int    `sql:"index"`
	CharacterName      string `sql:"not null;unique"`
	CharacterServer    string `sql:"not null"`
	CharacterLevel     int    `sql:"not null"`
	CharacterMain      bool   `sql:"not null"`
	CharacterClass     string `sql:"not null"`
	CharacterPic       string `sql:"not null"`
	CharacterSpec      string `sql:"not null"`
	CharacterCSSClass  string `sql:"not null"`
	CharacterTalentImg string `sql:"not null"`
}

// ForumMainPostsReplies is a table in the database
type ForumMainPostsReplies struct {
	gorm.Model
	ForumMainThreadsPostsID int `sql:"index"`
	UserID                  int `sql:"index"`
	ForumMainThreadsPosts   ForumMainThreadsPosts
	User                    User
	Body                    string
}

// ForumMainThreadsPosts is a table in the database
type ForumMainThreadsPosts struct {
	gorm.Model
	UserID             int `sql:"index"`
	ForumMainThreadsID int `sql:"index"`
	User               User
	ForumMainThreads   ForumMainThreads
	Title              string `sql:"not null;unique"`
	Body               string `sql:"type:text;not null"`
	Sticky             bool   `sql:"not null;default:0"`
	Views              int    `sql:"not null;default:0"`
	ReplyCount         int    `sql:"not null;default:0"`
	Replies            []ForumMainPostsReplies
}

// ForumMainThreads is the table in the database
type ForumMainThreads struct {
	gorm.Model
	ForumContainerID int    `sql:"index"`
	ThreadName       string `sql:"not null;unique"`
	Threads          int    `sql:"not null;default:0"`
	Replies          int    `sql:"not null;default:0"`
	AccessLevel      int    `sql:"not null;default:0"`
	Posts            []ForumMainThreadsPosts
}

// NewsFeed is the news that people will see when first
// visiting the page
type NewsFeed struct {
	gorm.Model
	UserID int `sql:"index"`
	User   User
	Title  string
	Body   string `sql:"type:text"`
}

func init() {
	var err error
	db, err = gorm.Open("mysql", "asd:asd@(asd:123)/testdb?parseTime=true")
	if err != nil {
		panic(err)
	}
	db.LogMode(true)

	// db.AutoMigrate(&User{}, &Characters{},
	// 	&ForumMainThreads{}, &ForumMainThreadsPosts{}, &ForumMainPostsReplies{}, &NewsFeed{})
}

func main() {

	posts := []ForumMainThreadsPosts{}
	if err := db.Order("created_at desc").Preload("User.Characters").Limit(10).Find(&posts).Error; err != nil {
		log.Println(err)
	}
	log.Printf("Posts: %+v\n\n\n", posts)
	for _, post := range posts {
		log.Printf("%+v", post.User.Characters)
	}

	log.Println("Finished")

}


```

Only the first element is getting their `Characters` populated:

```
[2016-01-10 15:59:52]  [245.46ms]  SELECT  * FROM `characters`  WHERE (`characters`.deleted_at IS NULL OR `characters`.deleted_at <= '0001-01-02') AND ((user_id IN ('48','5','30','13','51','31','4','5','5','5')))
```

**Comment by jinzhu** [comment]
Hi @Gacnt 

Get build exception when try to run the script. Could you submit another one?

**Comment by Gacnt** [comment]
@jinzhu Fixed the code block in my OP, didn't realize I was calling the structs from a different package so it compiled just fine :)

Should work now.

**Comment by bwiggs** [comment]
+1 Also having this issue.

**Comment by jinzhu** [comment]
Hi @Gacnt 

Could you create some default data or dump your test database and upload it here?

**Comment by Gacnt** [comment]
```
# ************************************************************
# Sequel Pro SQL dump
# Version 4096
#
# http://www.sequelpro.com/
# http://code.google.com/p/sequel-pro/
#
# Host: pugit.net (MySQL 5.5.46-0ubuntu0.14.04.2)
# Database: testdb
# Generation Time: 2016-01-13 00:39:13 +0000
# ************************************************************


/*!40101 SET @OLD_CHARACTER_SET_CLIENT=@@CHARACTER_SET_CLIENT */;
/*!40101 SET @OLD_CHARACTER_SET_RESULTS=@@CHARACTER_SET_RESULTS */;
/*!40101 SET @OLD_COLLATION_CONNECTION=@@COLLATION_CONNECTION */;
/*!40101 SET NAMES utf8 */;
/*!40014 SET @OLD_FOREIGN_KEY_CHECKS=@@FOREIGN_KEY_CHECKS, FOREIGN_KEY_CHECKS=0 */;
/*!40101 SET @OLD_SQL_MODE=@@SQL_MODE, SQL_MODE='NO_AUTO_VALUE_ON_ZERO' */;
/*!40111 SET @OLD_SQL_NOTES=@@SQL_NOTES, SQL_NOTES=0 */;


# Dump of table characters
# ------------------------------------------------------------

DROP TABLE IF EXISTS `characters`;

LOCK TABLES `characters` WRITE;
/*!40000 ALTER TABLE `characters` DISABLE KEYS */;

INSERT INTO `characters` (`id`, `created_at`, `updated_at`, `deleted_at`, `user_id`, `character_name`, `character_server`, `character_level`, `character_main`, `character_class`, `character_pic`, `character_spec`, `character_css_class`, `character_talent_img`)
VALUES
	(1,NULL,NULL,NULL,1,'asdfafsd','ASDFASDAF',100,1,'mage','','','',''),
	(2,NULL,NULL,NULL,2,'gdfxfbfdbbgxbbdxbxd','asfasfasfasvsavsvasvdasdva',100,1,'warrior','','','',''),
	(3,NULL,NULL,NULL,2,'avvssvasadvasdvasdv','asdvasdvasdvasdvasdvasvd',100,1,'priest','','','','');

/*!40000 ALTER TABLE `characters` ENABLE KEYS */;
UNLOCK TABLES;


# Dump of table forum_main_posts_replies
# ------------------------------------------------------------

DROP TABLE IF EXISTS `forum_main_posts_replies`;



# Dump of table forum_main_threads
# ------------------------------------------------------------

DROP TABLE IF EXISTS `forum_main_threads`;

LOCK TABLES `forum_main_threads` WRITE;
/*!40000 ALTER TABLE `forum_main_threads` DISABLE KEYS */;

INSERT INTO `forum_main_threads` (`id`, `created_at`, `updated_at`, `deleted_at`, `forum_container_id`, `thread_name`, `threads`, `replies`, `access_level`)
VALUES
	(1,NULL,NULL,NULL,NULL,'asafasfdasfd',0,0,0),
	(2,NULL,NULL,NULL,NULL,'zxcvzxcxzczx',0,0,0);

/*!40000 ALTER TABLE `forum_main_threads` ENABLE KEYS */;
UNLOCK TABLES;


# Dump of table forum_main_threads_posts
# ------------------------------------------------------------

DROP TABLE IF EXISTS `forum_main_threads_posts`;

LOCK TABLES `forum_main_threads_posts` WRITE;
/*!40000 ALTER TABLE `forum_main_threads_posts` DISABLE KEYS */;

INSERT INTO `forum_main_threads_posts` (`id`, `created_at`, `updated_at`, `deleted_at`, `user_id`, `forum_main_threads_id`, `title`, `body`, `sticky`, `views`, `reply_count`)
VALUES
	(1,NULL,NULL,NULL,1,1,'ll;lljj;j;jl','lfasfalsfdkasdfjlhasljdfhjasdhf',0,0,0),
	(2,NULL,NULL,NULL,2,2,'adsfs','adsfasdfasdsvsdavsdv',0,0,0),
	(3,NULL,NULL,NULL,1,1,'sfasdadadf','aasfasdfasfasfasdfasdf',0,0,0);

/*!40000 ALTER TABLE `forum_main_threads_posts` ENABLE KEYS */;
UNLOCK TABLES;


# Dump of table news_feeds
# ------------------------------------------------------------

DROP TABLE IF EXISTS `news_feeds`;



# Dump of table users
# ------------------------------------------------------------

DROP TABLE IF EXISTS `users`;

LOCK TABLES `users` WRITE;
/*!40000 ALTER TABLE `users` DISABLE KEYS */;

INSERT INTO `users` (`id`, `created_at`, `updated_at`, `deleted_at`, `username`, `email`, `password_hash`, `user_level`)
VALUES
	(1,NULL,NULL,NULL,'asdasd','asd@asd.com','asdasd',1),
	(2,NULL,NULL,NULL,'jlkljkjklklj','ljkl@dsgfsdfg.com','asdffsadfasdf',1);

/*!40000 ALTER TABLE `users` ENABLE KEYS */;
UNLOCK TABLES;



/*!40111 SET SQL_NOTES=@OLD_SQL_NOTES */;
/*!40101 SET SQL_MODE=@OLD_SQL_MODE */;
/*!40014 SET FOREIGN_KEY_CHECKS=@OLD_FOREIGN_KEY_CHECKS */;
/*!40101 SET CHARACTER_SET_CLIENT=@OLD_CHARACTER_SET_CLIENT */;
/*!40101 SET CHARACTER_SET_RESULTS=@OLD_CHARACTER_SET_RESULTS */;
/*!40101 SET COLLATION_CONNECTION=@OLD_COLLATION_CONNECTION */;
```

@jinzhu

**Comment by jinzhu** [comment]
@Gacnt seems invalid sql?

![image](https://cloud.githubusercontent.com/assets/6843/12281595/90f514b0-b9d2-11e5-801d-b3daa796c2ca.png)

**Comment by Gacnt** [comment]
@jinzhu I assumed you ran the auto-migrate in the `init` before importing the sql dump

**Comment by jinzhu** [comment]
It also doesn't work, maybe you could tried to run it by yourself.

**Comment by Gacnt** [comment]
@jinzhu updated the sql dump, should work now

**Comment by jinzhu** [comment]
![image](https://cloud.githubusercontent.com/assets/6843/12281730/89e80bcc-b9d3-11e5-8273-5baa869144ab.png)

**Comment by Gacnt** [comment]
jesus christ..... I wonder if this is related the Sequel Pro GUI, remove 

```

/*!40111 SET SQL_NOTES=@OLD_SQL_NOTES */;
/*!40101 SET SQL_MODE=@OLD_SQL_MODE */;
/*!40014 SET FOREIGN_KEY_CHECKS=@OLD_FOREIGN_KEY_CHECKS */;
/*!40101 SET CHARACTER_SET_CLIENT=@OLD_CHARACTER_SET_CLIENT */;
/*!40101 SET CHARACTER_SET_RESULTS=@OLD_CHARACTER_SET_RESULTS */;
/*!40101 SET COLLATION_CONNECTION=@OLD_COLLATION_CONNECTION */;

```

from the bottom of the sql dump, imported just fine for me

**Comment by Gacnt** [comment]
@jinzhu I just did a `mysqldump -u root -p testdb > test.sql` and its in the above code block, also I edited the code in the main post to better highlight the issue, just ran everything through a text and exported/reimported, everything seems to be in working order.

**Comment by jinzhu** [comment]
Hi @Gacnt 

I could reproduce your issue, thank you for your script.

**Comment by Gacnt** [comment]
You're welcome.

**Comment by Gacnt** [comment]
@jinzhu Do you have an ETA by chance on when you will troubleshoot/repair this issue or is it just going to be in the milestone

**Comment by jinzhu** [comment]
Hi @Gacnt 

Seems like not an easy to fix, so going to fix it in the v1.0 release #794

**Comment by Gacnt** [comment]
@jinzhu thanks

**Comment by jinzhu** [comment]
This has been fixed in the v1.0 branch, thank you for your report. #794
